command/lock.go

```diff
@@ -179,10 +160,10 @@ func (c *LockCommand) run(args []string, lu **LockUnlock) int {
 	}
 
 	// Setup the lock or semaphore
-	if limit == 1 {
-		*lu, err = c.setupLock(client, prefix, name, oneshot, wait, retry)
+	if c.limit == 1 {
+		*lu, err = c.setupLock(client, prefix, c.name, oneshot, c.timeout, c.monitorRetry)
```

Comment:
These ended up a little weird passing member variables to member functions. I'd probably make these all local variables.